### ISSUE TITLE
socket.timeout when running --http_headers

### ISSUE THREAD

**Issue by wondex** [nabla-c0d3/sslyze#499]
**Describe the bug**

```
Traceback (most recent call last):
  File "/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages/sslyze/scanner.py", line 264, in get_results
    result = implementation_cls.result_for_completed_scan_jobs(
  File "/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages/sslyze/plugins/http_headers_plugin.py", line 165, in result_for_completed_scan_jobs
    return completed_scan_jobs[0].result()
  File "/opt/rh/rh-python38/root/usr/lib64/python3.8/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/opt/rh/rh-python38/root/usr/lib64/python3.8/concurrent/futures/_base.py", line 388, in __get_result
    raise self._exception
  File "/opt/rh/rh-python38/root/usr/lib64/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages/sslyze/plugins/http_headers_plugin.py", line 184, in _retrieve_and_analyze_http_response
    http_response = HttpResponseParser.parse_from_ssl_connection(ssl_connection.ssl_client)
  File "/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages/sslyze/connection_helpers/http_response_parser.py", line 23, in parse_from_ssl_connection
    return cls._parse(ssl_conn.read)
  File "/opt/rh/rh-python38/root/usr/local/lib/python3.8/site-packages/sslyze/connection_helpers/http_response_parser.py", line 29, in _parse
    response = read_method(4096)
  File "/opt/rh/rh-python38/root/usr/local/lib64/python3.8/site-packages/nassl/ssl_client.py", line 223, in read
    encrypted_data = self._sock.recv(self._DEFAULT_BUFFER_SIZE)
socket.timeout: timed out
```

**To Reproduce**
Install SSLyze using: pip
Run the following command: /opt/rh/rh-python38/root/usr/bin/python3.8 -m sslyze --regular

**Expected behavior**
Successful scan

**Python environment (please complete the following information):**
OS: Red Hat Enterprise Linux Server release 7.9 (Maipo)
Python version: 3.8
SSLyzeL: 4.0.2

**Comment by nabla-c0d3** [comment]
@wondex Could you send me the server that triggered this issue? Otherwise I won't be able to fix it. Thanks!

**Comment by wondex** [comment]
Hi Alban,

This happens on multiple hosts and this are all Windows Server hosts with the IIS webserver.
Some have a redirect (302), but some doesn't.

Hope this hint could solve the issue.

**Comment by nabla-c0d3** [comment]
This error is visible there: https://github.com/nabla-c0d3/sslyze/runs/1944599074?check_suite_focus=true

**Comment by nabla-c0d3** [comment]
Fixed in [v4.0.4](https://github.com/nabla-c0d3/sslyze/releases/tag/4.0.4) for servers that don't speak HTTP. It doesn't address the situation where an IIS server will once in a while trigger a timeout although it does speak HTTP. I'm not sure there's anything SSLyze can do in that situation.
